lib/spack/spack/filesystem_view.py

```diff
@@ -125,6 +137,12 @@ def remove_standalone(self, spec):
         """
         raise NotImplementedError
 
+    def get_projection_for_spec(self, spec):
```

Comment:
I think this is fine but that any user of `YamlFilesystemView` that originally wanted to access `view.root` would now want to use this function. Therefore I think `view.root` should be replaced with `view._root` (i.e. that is now a private and internal detail of the view). Currently I don't think all build system implementations (e.g. `Python` and `Aspell`) are consistent with the other changes in this PR but making this change would help with that.

Reply:
It seems that the check in the `AspellDictPackage` class is still a correct use of the view root to check whether the view is being used for an extension or externally.

I've updated the `PythonPackage` build system to use the new syntax to get its location within the view. Since python packages are always extensions and extensions are linked into combinatorial views using non-combinatorial sub-views, it doesn't actually make a difference, but I think relying on that would be too fragile.

Reply:
I'm still not convinced that `root` shouldn't be something you're allowed to ask a view. But I don't think removing it will harm any of the current cases, so I'll make the change and we can always change it back later if necessary.